src/main/scala/io/qbeast/spark/index/SparkAutoIndexer.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.index
+
+import io.qbeast.core.model.AutoIndexer
+import org.apache.spark.ml.feature.OneHotEncoder
+import org.apache.spark.ml.feature.StringIndexer
+import org.apache.spark.ml.feature.VectorAssembler
+import org.apache.spark.ml.linalg.Matrix
+import org.apache.spark.ml.stat.Correlation
+import org.apache.spark.ml.Pipeline
+import org.apache.spark.qbeast.config.MAX_NUM_COLUMNS_TO_INDEX
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.functions.unix_timestamp
+import org.apache.spark.sql.DataFrame
+
+object SparkAutoIndexer extends AutoIndexer[DataFrame] with Serializable {
+
+  /**
+   * The maximum number of columns to index.
+   *
+   * @return
+   */
+  override def MAX_COLUMNS_TO_INDEX: Int = MAX_NUM_COLUMNS_TO_INDEX
+
+  /**
+   * Chooses the columns to index without limiting the number of columns.
+   *
+   * @param data
+   *   the data to index
+   * @return
+   */
+  override def chooseColumnsToIndex(data: DataFrame): Seq[String] =
+    chooseColumnsToIndex(data, MAX_COLUMNS_TO_INDEX)
+
+  override def chooseColumnsToIndex(data: DataFrame, numColumnsToIndex: Int): Seq[String] = {
+
+    // IF there's no data to write, we return all the columns to index
+    if (data.isEmpty) {
+      return data.columns.take(numColumnsToIndex)
+    }
+
+    var updatedData = data
+    val inputCols = data.columns
+
+    // Convert timestamp columns to Unix timestamps and update the DataFrame
+    val timestampCols = inputCols.filter(colName =>
+      data.schema(colName).dataType == org.apache.spark.sql.types.TimestampType)
+    timestampCols.foreach { colName =>
```

Comment:
Scala zealots like to chain two closures, so the local variable timestampCols is redundant.